commit 0158021c21657bdc537895a634d1a73b593e6639
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-05-03T18:29:31-07:00 (5 months ago)
    
    joyent/smartos-live#771 imgadm vacuum will try to reap images used by bhyve vms
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/src/img/CHANGES.md b/src/img/CHANGES.md
index 31fd9b1e..52b63c41 100644
--- a/src/img/CHANGES.md
+++ b/src/img/CHANGES.md
@@ -5,6 +5,10 @@ Known issues:
 - Docker image imports are experimental. Docker image import also only supports
   Docker Registry v2.
 
+## 3.9.3
+
+- joyent/smartos-live#771 imgadm vacuum will try to reap images used by bhyve vms
+
 ## 3.9.2
 
 - TRITON-622 'stdin.write' in promptYesNo func in common.js breaks on node v10
diff --git a/src/img/lib/imgadm.js b/src/img/lib/imgadm.js
index d9430658..c60fb9fd 100644
--- a/src/img/lib/imgadm.js
+++ b/src/img/lib/imgadm.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2018, Joyent, Inc. All rights reserved.
+ * Copyright 2019 Joyent, Inc.
  *
  * * *
  * The main imgadm functionality. The CLI is a light wrapper around this tool.
@@ -88,7 +88,15 @@ var DEFAULT_CONFIG = {};
 var SET_REQUIREMENTS_BRAND_BRANDS = ['bhyve', 'lx', 'kvm'];
 
 /* BEGIN JSSTYLED */
-var VMADM_FS_NAME_RE = /^([a-zA-Z][a-zA-Z\._-]*)\/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})(-disk\d+)?$/;
+/**
+ * For vm filesystems, we have three variations to which an image can be
+ * cloned to:
+ *  1. /zones/<uuid>        (smartos, lx and docker)
+ *  2. /zones/<uuid>-diskN  (kvm)
+ *  3. /zones/<uuid>/diskN  (bhyve)
+ */
+var VMADM_FS_NAME_RE = /^([a-zA-Z][a-zA-Z\._-]*)\/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})([-/]disk\d+)?$/;
+
 var VMADM_IMG_NAME_RE = /^([a-zA-Z][a-zA-Z\._-]*)\/([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})$/;
 var UUID_RE = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/;
 /* END JSSTYLED */
diff --git a/src/img/package.json b/src/img/package.json
index 73a3cd43..a00e0c3c 100644
--- a/src/img/package.json
+++ b/src/img/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgadm",
   "description": "Manage SmartOS virtual machine images.",
-  "version": "3.9.2",
+  "version": "3.9.3",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
